libgadget/hydra.c

```diff
@@ -214,9 +217,12 @@ hydro_force(const ActiveParticles * act, int WindOn, const double hubble, const
     {
         priv->gravkicks[i] = get_exact_gravkick_factor(CP, times.Ti_kick[i], times.Ti_Current);
         priv->hydrokicks[i] = get_exact_hydrokick_factor(CP, times.Ti_kick[i], times.Ti_Current);
+        /* For density: last active drift time is Ti_kick - 1/2 timestep as the kick time is half a timestep ahead.
+         * For active particles no density drift is needed.*/
+        if(!is_timebin_active(i, times.Ti_Current))
+            priv->drifts[i] = get_exact_drift_factor(CP, times.Ti_kick[i] - dti_from_timebin(i)/2, times.Ti_Current);
```

Comment:
So the values in these kicks and drifts table are to predict the quantity at Ti_Current for a particle at time bin i?

Should we track times.Ti_drift[i] as times.Ti_kick[i] - dti_from_timebin(i) / 2 ?

Reply:
A good idea. Perhaps Ti_lastactivedrift[i]?